clippy_lints/src/matches/collapsible_match.rs

```diff
@@ -108,6 +108,28 @@ fn check_arm<'tcx>(
                 "the outer pattern can be modified to include the inner pattern",
             );
         });
+    } else if let Some(inner) = IfLetOrMatch::parse(cx, inner_expr)
```

Comment:
The conditional branch added this time could have been written by nesting `if`s together with the conditional branch above, but I have not done so for the sake of readability.